modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/utils/OpenApiUtil.java

```diff
@@ -0,0 +1,346 @@
+     */
+    public String getOpenApiUrl(RegisterServiceEntity registerServiceEntity) {
+        if (Objects.isNull(registerServiceEntity)) {
+            throw new IllegalArgumentException("Registered service is null.");
+        }
+        String id = registerServiceEntity.getId().toString();
+        File file = new File(getOpenApiDir(), id + ".html");
+        if (file.exists()) {
+            return getOpenApiUrl() + id + ".html";
+        } else {
+            return creatServiceApi(registerServiceEntity);
+        }
+    }
+
+    /**
+     * Get serviceUrl.
+     *
+     * @return serviceUrl
+     */
+    private String getOpenApiUrl() {
+        String ip = "localhost";
+        try {
+            InetAddress address = InetAddress.getLocalHost();
+            ip = address.getHostAddress();
+        } catch (UnknownHostException e) {
+            log.error("Get localHost error,");
+        }
+        return HTTP + ip + ":" + port + "/" + openapiPath;
+    }
+
+    /**
+     * Get openApi file path.
+     *
+     * @return openApiUrl
+     */
+    private String getOpenApiDir() {
+        String rootPath = System.getProperty("user.dir");
+        return rootPath + File.separator + openapiPath;
+    }
+
+    /**
+     * update OpenApi for registered service .
+     *
+     * @param registerService Registered services.
+     */
+    public void updateServiceApi(RegisterServiceEntity registerService) {
+        File file = new File(getOpenApiDir(), registerService.getId() + ".html");
+        if (file.exists()) {
+            log.info("Delete old openapi file:{}, success:{}", file.getName(), file.delete());
+        }
+        creatServiceApi(registerService);
+    }
+
+    /**
+     * create OpenApi for registered service .
+     *
+     * @param registerService Registered services.
+     */
+    public String creatServiceApi(RegisterServiceEntity registerService) {
+        List<DeployVariable> deployVariables = registerService.getOcl().getDeployment()
+                .getContext();
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        String propertiesStr = null;
+        String requiredStr = null;
+        try {
+            propertiesStr =
+                    mapper.writeValueAsString(
+                            deployVariableValidator.getVariableApiInfoMap(deployVariables)).replace(
+                            "\"[", "[").replace("]\"", "]").replace("\\", "").replace("'", "");
+            requiredStr = mapper.writeValueAsString(
+                    deployVariableValidator.getRequiredKeySet(deployVariables));
+        } catch (JsonProcessingException e) {
+            log.error("Failed to write value as string.", e);
+        }
+        String apiJson = String.format("""
+                {
+                    "openapi": "3.0.1",
+                    "info": {
+                      "title": "OpenAPI definition",
+                      "version": "v0"
```

Comment:
can this be the service version?

Reply:
Yes. I will fill the field 'version' with service version.